core/src/memory/mod.rs

```diff
@@ -3,23 +3,20 @@ pub mod memory_models;
 pub mod symbolic;
 pub mod symbolic_bv;
 pub mod r#abstract;
-use std::marker::PhantomData;
 
-use crate::instructions::val::Val;
 use error::MemoryError;
-use z3::{ast::Array, FuncDecl};
 
-pub type MemoryResult<T> = Result<T, MemoryError>;
-pub trait ReadOnlyMem: Sized {
+pub trait Mem: Sized {
     type MemVal;
     type Index;
-    fn read(&self, idx: Self::Index) -> MemoryResult<Option<Self::MemVal>>;
 }
 
```

Comment:
Added Mem trait. Useful like in the case of ADD where memory isn't touched

Reply:
Also allows machine/instruction trait to be less restrictive (not require RWMem)